x/nexus/exported/types.go

```diff
@@ -6,6 +6,61 @@ import (
 	sdk "github.com/cosmos/cosmos-sdk/types"
 )
 
+// Validate validates the TransferState
+func (m TransferState) Validate() error {
+	_, ok := TransferState_name[int32(m)]
+	if !ok {
+		return fmt.Errorf("unknown transfer state")
+	}
+
+	if m == TRANSFER_STATE_UNSPECIFIED {
+		return fmt.Errorf("unspecified transfer state")
+	}
+
+	return nil
+}
+
+// Validate validates the CrossChainTransfer
+func (m CrossChainTransfer) Validate() error {
+	if err := m.Recipient.Validate(); err != nil {
+		return err
+	}
+
+	if err := m.Asset.Validate(); err != nil {
+		return err
+	}
+
+	if err := m.State.Validate(); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// Validate validates the CrossChainAddress
+func (m CrossChainAddress) Validate() error {
+	if err := m.Chain.Validate(); err != nil {
+		return err
+	}
+
+	// TODO: validate .Address according to .Chain
```

Comment:
I don't think this will be trivial, because the validation callbacks are are stored on the keeper, using the same methodology that we use for routing signatures to their respective modules.

Reply:
agreed, could this be done during import? Either way remove this todo here, there is nothing we can do about it here